samples/tab-sso/src/nodejs/src/components/Tab.js

```diff
@@ -2,166 +2,161 @@
       failureCallback: (error) => {this.ssoLoginFailure(error)}
     };
 
     microsoftTeams.authentication.getAuthToken(authTokenRequestOptions);
-  }  
+  }
+
+  ssoLoginSuccess = async (result) => {
+    this.setState({ssoToken: result});
 
-  ssoLoginSuccess = async (token) => {
-	this.setState({ssoToken:token});
-	let decodedToken = jwt_decode(token);
-	this.setState({
-		name: decodedToken.name,
-		aadObjectId: decodedToken.oid,
-		upn: decodedToken.upn,
-		tenantId: decodedToken.tid
-	});
-    this.exchangeClientTokenForServerToken(token);
+    // The result variable is the SSO token
+    this.exchangeClientTokenForServerToken(result);
   }
 
-  ssoLoginFailure(error){
-    console.error("SSO failed: ",error);
-    this.setState({error:true});
+  ssoLoginFailure = (error) => {
+    console.error("SSO failed: ", error);
+    this.setState({error: true});
   }
 
-  //Exchange the SSO access token for a Graph access token
-  //Learn more: https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow
+  // Exchange the SSO access token for a Graph access token
+  // Learn more: https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow
   exchangeClientTokenForServerToken = async (token) => {
 
-    let response = await fetch('/getGraphAccessToken?ssoToken='+token).catch(this.unhandledFetchError); //This calls getGraphAccessToken route in /api-server/app.js
+    let serverURL = `/getGraphAccessToken?ssoToken=${token}`;
+    let response = await fetch(serverURL).catch(this.unhandledFetchError); //This calls getGraphAccessToken route in /api-server/app.js
     let data = await response.json().catch(this.unhandledFetchError);
 
-    if(!response.ok && data.error==='consent_required'){
-      //A consent_required error means it's the first time a user is logging into to the app, so they must consent to sharing their Graph data with the app.
-      //They may also see this error if MFA is required. Proceed to show the consent dialogue.
-      this.setState({consentRequired:true}); //This displays the consent button.
+    // Show a popup dialogue prompting the user to consent to the required API permissions.
+    // Learn more: https://docs.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/authentication/auth-tab-aad#initiate-authentication-flow
+    if (!response.ok && data.subError === 'consent_required') {
+
+      // A consent_required error means it's the first time a user is logging into to the app, so they must 
+      // consent to sharing their Graph data with the app. They may also see this error if MFA is required.
+      this.setState({consentRequired: true});
+      
+      try {
+
+        // Access token acquisition. Visit: 
+        // https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/acquire-token.md
+        const response = await this.props.msalContext.instance.acquireTokenPopup({

```

Comment:
This approach wont work, as it will be blocked on desktop teams. You need to keep using the `microsoftTeams.authentication.authenticate` API (with the consent and close popup pages), which when opened in the popup created by teams, invokes loginRedirect and then processes the redirect in the popup and passes it back.

Reply:
1. In tab app, check if login/tokens is needed (acquireTokenSilent)
2. If so, invoke `microsoftTeams.authentication.authenticate` pointing to a page that invokes `loginRedirect` on page load.
3. Set redirect uri to page that will parse the response via `handleRedirectPromise` and then pass it back via `microsoftTeams.authentication.notifySuccess` or `microsoftTeams.authentication.notifyFailure`.

This is essentially what exists in this sample already, it can just be updated to use new versions of MSAL.

Reply:
Of course.. 🤦 I'll convert PR to draft and work on this